### PR TITLE
impl TryFrom for Address

### LINKED ISSUE #248: Bech32 improvements

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

impl TryFrom for Address and remove Bech32Address

## Links to any relevant issues

Fixes https://github.com/iotaledger/bee/issues/248

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

try_into test and tested with iota.rs

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-message/src/address/mod.rs [hunk 1]
```diff
@@ -67,22 +67,14 @@ impl From<Ed25519Address> for Address {
 impl TryFrom<&str> for Address {
```
Comment (thibault-martinez) [inline bee-message/src/address/mod.rs]: This is actually `FromStr`, better to implement this one instead

File: bee-message/src/address/mod.rs [hunk 2]
```diff
@@ -67,22 +67,14 @@ impl From<Ed25519Address> for Address {
...
-        if let Ok(address) = Ed25519Address::from_str(&address) {
-            Ok(Address::Ed25519(address))
-        } else {
-            Address::try_from_bech32(address)
-        }
+        Address::try_from_bech32(address)
     }
 }
 
 impl TryFrom<String> for Address {
```
Comment (thibault-martinez) [inline bee-message/src/address/mod.rs:74]: This can just call the other one

Comment (Thoralf-M) [reply bee-message/src/address/mod.rs:74]: Correct now?

Comment (thibault-martinez) [reply bee-message/src/address/mod.rs:74]: Yes sir

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**